frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -961,6 +987,79 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Sets the beneficiary of staking rewards.
+        ///
+        /// The dispatch origin is staker.
+        #[pallet::weight(10_000)]
+        pub fn set_rewards_beneficiary(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+            beneficiary: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            ensure!(
+                Self::is_active(&contract_id),
+                Error::<T>::NotOperatedContract
+            );
+            ensure!(beneficiary != staker, Error::<T>::InvalidBeneficiary);
+
+            RewardsBeneficiary::<T>::insert(&staker, &contract_id, &beneficiary);
+
+            Self::deposit_event(Event::<T>::BeneficiarySet(staker, contract_id, beneficiary));
+            Ok(().into())
+        }
+
+        /// Removes the beneficiary of staking rewards.
+        ///
+        /// The dispatch origin is staker.
+        #[pallet::weight(10_000)]
+        pub fn remove_rewards_beneficiary(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            RewardsBeneficiary::<T>::remove(&staker, &contract_id);
+
+            Self::deposit_event(Event::<T>::BeneficiaryRemoved(staker, contract_id));
+            Ok(().into())
+        }
+
+        /// Updates the beneficiary to a new account.
+        ///
+        /// The dispatch origin is the current beneficiary.
+        #[pallet::weight(10_000)]
+        pub fn update_rewards_beneficiary(
```

Comment:
Perhaps `delegate_rewards_beneficiary`?
Since this cannot be used by the staker, so it's more clear.